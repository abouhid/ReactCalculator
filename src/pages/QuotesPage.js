import React from 'react';
import quotesdata from '../data/quotesdata';

const Quotes = () => {
  const quoteList = quotesdata.map(item => (

    <div className="quote-container" key={item.id}>
      <img alt="img" src={item.src} />
      <div className="side-text ">

        <h2>
          &quot;
          {item.quote}
          &quot;
        </h2>
        <h4>
          -
          {item.author}
        </h4>
      </div>

    </div>
  ));
  return (

    <div className="App">
      {quoteList}
    </div>
  );
};

export default Quotes;
