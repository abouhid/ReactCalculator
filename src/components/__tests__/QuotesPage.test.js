import React from 'react';
import renderer from 'react-test-renderer';
import QuotesPage from '../../pages/QuotesPage';

describe('QuotesPage', () => {
  it('Tests the rendering of components', () => {
    const tree = renderer
      .create(<QuotesPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});