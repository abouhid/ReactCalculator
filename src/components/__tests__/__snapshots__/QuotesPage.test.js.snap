// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotesPage Tests the rendering of components 1`] = `
<div
  className="App"
>
  <div
    className="quote-container"
  >
    <img
      alt="img"
      src="https://images.gr-assets.com/authors/1526204227p2/42775.jpg"
    />
    <div
      className="side-text "
    >
      <h2>
        "
        Solving a problem for which you know there’s an answer is like climbing a mountain with a guide, along a trail someone else has laid. In mathematics, the truth is somewhere out there in a place no one knows, beyond all the beaten paths. And it’s not always at the top of the mountain. It might be in a crack on the smoothest cliff or somewhere deep in the valley.
        "
      </h2>
      <h4>
        -
        Yoko Ogawa, The Housekeeper and the Professor
      </h4>
    </div>
  </div>
  <div
    className="quote-container"
  >
    <img
      alt="img"
      src="https://images.gr-assets.com/authors/1429114964p2/9810.jpg"
    />
    <div
      className="side-text "
    >
      <h2>
        "
        Since the mathematicians have invaded the theory of relativity I do not understand it myself any more.
        "
      </h2>
      <h4>
        -
        Albert Einstein
      </h4>
    </div>
  </div>
  <div
    className="quote-container"
  >
    <img
      alt="img"
      src="https://i.kym-cdn.com/photos/images/original/001/318/553/497.jpg"
    />
    <div
      className="side-text "
    >
      <h2>
        "
        The risk I took was calculated, but man, am I bad at math.
        "
      </h2>
      <h4>
        -
        small Kestrel
      </h4>
    </div>
  </div>
</div>
`;

exports[`QuotesPage component renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="quote-container"
  >
    <img
      alt="img"
      src="https://images.gr-assets.com/authors/1526204227p2/42775.jpg"
    />
    <div
      className="side-text "
    >
      <h2>
        "
        Solving a problem for which you know there’s an answer is like climbing a mountain with a guide, along a trail someone else has laid. In mathematics, the truth is somewhere out there in a place no one knows, beyond all the beaten paths. And it’s not always at the top of the mountain. It might be in a crack on the smoothest cliff or somewhere deep in the valley.
        "
      </h2>
      <h4>
        -
        Yoko Ogawa, The Housekeeper and the Professor
      </h4>
    </div>
  </div>
  <div
    className="quote-container"
  >
    <img
      alt="img"
      src="https://images.gr-assets.com/authors/1429114964p2/9810.jpg"
    />
    <div
      className="side-text "
    >
      <h2>
        "
        Since the mathematicians have invaded the theory of relativity I do not understand it myself any more.
        "
      </h2>
      <h4>
        -
        Albert Einstein
      </h4>
    </div>
  </div>
  <div
    className="quote-container"
  >
    <img
      alt="img"
      src="https://i.kym-cdn.com/photos/images/original/001/318/553/497.jpg"
    />
    <div
      className="side-text "
    >
      <h2>
        "
        The risk I took was calculated, but man, am I bad at math.
        "
      </h2>
      <h4>
        -
        small Kestrel
      </h4>
    </div>
  </div>
</div>
`;
